deriva/core/ermrest_model.py

```diff
@@ -281,6 +281,97 @@ def ktup(k):
+          :param fkey_defs: a list of ForeignKey.define() results for foreign key definitions
+          :param comment: a comment string for the table
+          :param acls: a dictionary of ACLs for specific access modes
+          :param acl_bindings: a dictionary of dynamic ACL bindings
+          :param annotations: a dictionary of annotations
+          :param provide_system: whether to inject standard system column definitions when missing from column_defs
+
+          These core asset table columns are generated automatically if
+          absent from the input column_defs.
+
+          - Filename: ermrest_curie, unique not null, default curie template "%s:{RID}" % curie_prefix
+          - URL: Location of the asset, unique not null.  Default template is:
+                    /hatrac/tname/{{{_URL.Filename}}}.{{{_URL.MD5}}} where tname is the name of the asset table.
+          - Length: Length of the asset.
+          - MD5: text
+          - Description: markdown, not null
+
+          However, caller-supplied definitions override the default.
+
+          In addition to creating the columns, this function also creates an asset annotation on the URL column to
+          facilitate use of the table by Chaise.
+          """
+
+        if not hatrac_template:
+            hatrac_template='/hatrac/%s/%s/{{#encode}}{{{Filename}}}{{/encode}}.{{{MD5}}}' % (sname, tname)
+
+        def add_asset_annotations(custom):
+            annotations =  {_ec.tag['table_display']: {'row_name': {'row_markdown_pattern': '{{{Filename}}}'}}}
+            annotations.update(custom)
+            return annotations
+
+        def add_asset_columns(custom):
+            return [
+                       col_def
+                       for col_def in [
+                    Column.define('URL', builtin_types['text'],
+                                  nullok=False,
+                                  annotations={
+                                      _ec.tag.asset: {
+                                          'filename_column': 'Filename',
+                                          'byte_count_column': 'Length',
+                                          'url_pattern': hatrac_template,
+                                          'md5': 'MD5'
+                                      }
+                                  },
+                                  comment='URL to the asset'),
+                    Column.define('Filename', builtin_types['text'],
+                                  comment='Filename of the asset that was uploaded'),
+                    Column.define('Description', builtin_types['markdown'],
+                                  comment='Description of the asset'),
+                    Column.define('Length', builtin_types['int8'], nullok=False, comment='Asset length (bytes)'),
+                    Column.define('MD5', builtin_types['text'], nullok=False)
```

Comment:
should add a `comment` here since all the rest of your columns have one.

Reply:
Correct.  I would like to have a way of naming the URL that will always work.  I'm open to other suggestions.

The %s in the previous is the table name.  So I'm assuming that you would store assets in a directory that has the name of the asset table in the path.